spec/expression.dd

```diff
@@ -770,6 +766,12 @@ $(GNAME DeleteExpression):
         is called.
     )
 
+    $(UNDEFINED_BEHAVIOR
+    $(OL
+    $(LI Using `delete` to free memory allocated by the garbage collector.)
```

Comment:
Isn't there a "not" missing?
Using `delete` with GC allocated objects is well-defined, e.g. https://github.com/dlang/druntime/blob/master/src/rt/lifetime.d#L177

Reply:
Look closely at the link - it free's the data. There could still be other copies of the pointer.

Reply:
1) This is different to the existing statement that you removed:

> If the garbage collector was __not__ used to allocate the memory for the instance, undefined behavior will result.

2) Is `GC.free` on GC-allocated objects really undefined behavior?
AFAICT what it exactly does depends on the GC implementation, but currently it's `freeNoSync` which tries to free the memory if possible.

https://github.com/dlang/druntime/blob/master/src/gc/impl/conservative/gc.d#L818

Anyhow, `delete` is luckily finally deprecated, so we hopefully can remove it entirely soon.